parsec/backend/client_context.py

```diff
@@ -0,0 +1,162 @@
+# Parsec Cloud (https://parsec.cloud) Copyright (c) AGPLv3 2019 Scille SAS
+
+import trio
+from typing import Optional
+
+from parsec.crypto import VerifyKey, PublicKey
+from parsec.api.version import ApiVersion
+from parsec.api.transport import Transport
+from parsec.api.protocol import (
+    ServerHandshake,
+    OrganizationID,
+    UserID,
+    DeviceName,
+    DeviceID,
+    HumanHandle,
+)
+from parsec.backend.invite import Invitation
+
+
+class BaseClientContext:
+    __slots__ = ("transport", "handshake")
+
+    def __init__(self, transport: Transport, handshake: ServerHandshake):
+        self.transport = transport
+        self.handshake = handshake
+
+    @property
+    def api_version(self) -> ApiVersion:
+        return self.handshake.backend_api_version
+
+    @property
+    def handshake_type(self) -> str:
+        return self.handshake.answer_type
+
+
+class AuthenticatedClientContext(BaseClientContext):
+    __slots__ = (
+        "organization_id",
+        "device_id",
+        "human_handle",
+        "is_admin",
+        "public_key",
+        "verify_key",
+        "event_bus_ctx",
+        "channels",
+        "realms",
+        "conn_id",
+        "logger",
+    )
+
+    def __init__(
+        self,
+        transport: Transport,
+        handshake: ServerHandshake,
+        organization_id: OrganizationID,
+        device_id: DeviceID,
+        human_handle: Optional[HumanHandle],
+        is_admin: bool,
+        public_key: PublicKey,
+        verify_key: VerifyKey,
+    ):
+        super().__init__(transport, handshake)
+        self.organization_id = organization_id
+        self.is_admin = is_admin
+        self.device_id = device_id
+        self.human_handle = human_handle
+        self.public_key = public_key
+        self.verify_key = verify_key
+
+        self.event_bus_ctx = None  # Overwritten in BackendApp.handle_client
+        self.channels = trio.open_memory_channel(100)
```

Comment:
Why 100? Why not `math.inf`?

Reply:
This channel is consumed by the client, hence we don't want to flood the server in case the client subscribe to events but then doesn't listen to them